docs/adr/adr-009-public-api.md

```diff
@@ -0,0 +1,438 @@
+# ADR 009: Public API
+
+## Authors
+
+@Wondertan @renaynay
+
+## Changelog
+
+- 2022-03-08: initial version
+- 2022-08-03: additional details
+
+## Context
+
+Celestia Node has been built for almost half a year with a bottom-up approach to
+development. The core lower level components are built first and public API
+around them is getting organically shaped. Now that the project is maturing and
+its architecture is better defined, it's a good time to formally define a set of
+modules provided by the node and their respective APIs.
+
+## Alternative Approaches
+
+### Node Type centric Design
+
+Another discussed approach to defining an API could be an onion style, where
+each new layer is a feature set of a broader node type. However, it turns out
+that this approach does not seem to match the reality we have, where each node
+type implements all the possible APIs but with the variable implementations
+matching resource constraints of a type.
+
+## Design
+
+### Goals
+
+- Ergonomic. Simple, idiomatic and self-explanatory.
+- Module centric(modular). The API is not monolithic and is segregated into
```

Comment:
nit/non-blocking:
```suggestion
- Module centric (modular). The API is not monolithic and is segregated into
```

I see this a lot in other documentation and comments fwiw.